### PR TITLE
Feature/311 - MetadataImpl isEmpty request

### PR DESCRIPTION
**Pull request by davidjgonzalez**
oops - i never hit "Create PR" the first time :(

- Added tests for this method (removed Junit5 dep, added mockito)
- Handle Collections
- Genericized Array handling

### CODE REVIEW

File: core/src/main/java/com/adobe/aem/commons/assetshare/components/search/impl/StatisticsImpl.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ public long getTotal() {
 
     @Override
     public boolean hasMore() {
-        return search.getResults().isMore();
+        return search.getResults().isMoreThanTotal();
```
Comment (HitmanInWis) [inline core/src/main/java/com/adobe/aem/commons/assetshare/components/search/impl/StatisticsImpl.java:58]: Was this intended for this PR?

File: core/src/main/java/com/adobe/aem/commons/assetshare/search/results/Results.java [hunk 2]
```diff
@@ -54,7 +54,14 @@
     long getTotal();
 
     /**
-     * @return true if there are more results. false if there are no more results.
+     * @return value of has more from search results. This method that not takes into account that guessTotal is bigger then all results
```
Comment (HitmanInWis) [inline core/src/main/java/com/adobe/aem/commons/assetshare/search/results/Results.java:57]: Was this intended for this PR?

File: core/src/main/java/com/adobe/aem/commons/assetshare/search/results/impl/results/QueryBuilderResultsImpl.java [hunk 3]
```diff
@@ -46,6 +46,7 @@ public QueryBuilderResultsImpl(List<Result> results,
         this.nextOffset = searchResult.getNextPage() != null ? searchResult.getNextPage().getStart() : -1;
         this.runningTotal = searchResult.getStartIndex() + searchResult.getHits().size();
 
+        this.moreThanTotal = searchResult.hasMore();
```
Comment (HitmanInWis) [inline core/src/main/java/com/adobe/aem/commons/assetshare/search/results/impl/results/QueryBuilderResultsImpl.java:49]: Was this intended for this PR?

File: core/src/main/java/com/adobe/aem/commons/assetshare/components/details/impl/MetadataImpl.java [hunk 4]
```diff
@@ -152,7 +154,19 @@ public boolean isEmpty() {
...
             return true;
         } else {
-            return combinedProperties.get(getPropertyName()) == null;
+            Object val = combinedProperties.get(getPropertyName());
+            if (val == null) {
+                return true;
+            } else if (val instanceof String) {
+                return StringUtils.isBlank((String) val);
+            } else if (val instanceof Object[]) {
+                return ArrayUtils.isEmpty((Object[]) val);
```
Comment (HitmanInWis) [inline core/src/main/java/com/adobe/aem/commons/assetshare/components/details/impl/MetadataImpl.java:163]: I actually had a use case for a client where there was a property that was a String[] but the first value was empty.  Ideally we can still handle that case as well.

**Review by HitmanInWis — CHANGES_REQUESTED** [review]

**Comment by davidjgonzalez** [comment]
ugh - forgot to switch branches...

will try again before work tomorrow

**Pull request closed without merging**